src/utils/components/DiskModal/DiskModal.tsx

```diff
@@ -32,17 +33,19 @@ const DiskModal: FC<V1DiskModalProps> = ({
   const Modal = modalsBySource[createDiskSource || editDiskSource];
 
   return (
-    <Modal
-      createDiskSource={createDiskSource}
-      editDiskName={editDiskName}
-      isCreated={!isEmpty(createdPVCName)}
-      isOpen={isOpen}
-      onClose={onClose}
-      onSubmit={onSubmit}
-      onUploadedDataVolume={onUploadedDataVolume}
-      pvc={pvc}
-      vm={vm}
-    />
+    <WizardVMContextProvider>
```

Comment:
I believe the disk modal is general, why adding the wizard context to it?

Reply:
I need access to the registry credentials in `RegistryDiskModal`. I have the `useRegistryCredentials` hook for that, but it requires access to the context. It was either this, prop drill the credentials, or create a special component and muddy the modal selection code.

I'm open to other suggestions if you have them.

Reply:
The Diskmodal has `onSubmit` callback that you can use to update the credentials. Is this a good suggestion?